### PR TITLE
Test coverage

### PR DESCRIPTION
**Pull request by fmontei**
This commit adds the initial unit testing framework for Deckhand.
Included in this commit:
      * tox job for py27 and py35 unit tests
      * necessary testing requirements in test-requirements.txt
      * a few unit tests for core api framework

Future tests will be written once the framework becomes more fleshed out.

### CODE REVIEW

**Review by alanmeadows — APPROVED** [review]

File: tox.ini [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+                      flake8
 setenv = VIRTUAL_ENV={envdir}
+         OS_TEST_PATH=./deckhand/tests/unit
          LANGUAGE=en_US
          LC_ALL=en_US.utf-8
-deps=
-  -r{toxinidir}/requirements.txt
-  -r{toxinidir}/test-requirements.txt
-whitelist_externals = flake8
+passenv = OS_STDOUT_CAPTURE OS_STDERR_CAPTURE OS_TEST_TIMEOUT OS_TEST_LOCK_PATH OS_TEST_PATH http_proxy HTTP_PROXY https_proxy HTTPS_PROXY no_proxy NO_PROXY
```
Comment (mark-burnett) [inline tox.ini:15]: I think this would be more readable and easier to update in multi-line style.

**Review by mark-burnett — APPROVED** [review]
Looks good @fmontei

**Comment by mark-burnett** [comment]
@fmontei I'm happy to merge this once it's out of conflict.

File: tox.ini (hunk 1 above)
Comment (fmontei) [reply tox.ini:15]: Fixed in follow-up commit.

**Comment by fmontei** [comment]
@mark-burnett Thanks. This has already been merged in another PR. Going to abandon this.

**Pull request closed without merging**